src/ImageSharp/Formats/ImageDecoderUtilities.cs

```diff
@@ -7,12 +7,55 @@
 using SixLabors.ImageSharp.IO;
 using SixLabors.ImageSharp.Memory;
 using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
 
 namespace SixLabors.ImageSharp.Formats
 {
+    /// <summary>
+    /// Utility methods for <see cref="IImageDecoder"/>.
+    /// </summary>
     internal static class ImageDecoderUtilities
     {
-        public static IImageInfo Identify(
+        /// <summary>
+        /// Performs a resize operation against the decoded image. If the target size is not set, or the image size
+        /// already matches the target size, the image is untouched.
+        /// </summary>
+        /// <param name="options">The decoder options.</param>
+        /// <param name="image">The decoded image.</param>
+        public static void Resize(DecoderOptions options, Image image)
+        {
+            if (ShouldResize(options, image))
+            {
+                ResizeOptions resizeOptions = new()
+                {
+                    Size = options.TargetSize.Value,
+                    Sampler = KnownResamplers.Box,
```

Comment:
I believe we should make resampler type configurable to mimic MagicScaler functionality.

Reply:
I guess so, could easily add it to the general options. My original thinking though was that if they wanted more control they could use the processor.

Reply:
I've added it.

Reply:
> My original thinking though was that if they wanted more control they could use the processor.

Totally missed that possibility, gonna need some sleep.
But I think all-in-one API may still be superior than two-step API.
Would like to hear Anton on this subject.

Reply:
The the target-size decoder is a memory-optimized version of the two step decode-resize workflow. In an ideal world I would feature-match them, so we don't force the user to choose between perf and flexibility. However I understand that it also makes sense to keep `DecoderOptions` as simple as possible. I think adding Resampler makes sense, we'll se if users need more.